zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -397,8 +399,39 @@ impl<'a, F: Field> EVMConstraintBuilder<'a, F> {
         self.query_cell_with_type(CellType::LookupByte)
     }
 
-    pub(crate) fn query_word_rlc<const N: usize>(&mut self) -> RandomLinearCombination<F, N> {
-        RandomLinearCombination::<F, N>::new(self.query_bytes(), self.challenges.evm_word())
+    // TODO remove me
+    // pub(crate) fn query_word_rlc<const N: usize>(&mut self) -> RandomLinearCombination<F, N> {
+    //     RandomLinearCombination::<F, N>::new(self.query_bytes(), self.challenges.evm_word())
+    // }
+
+    // default query_word is 2 limbs
```

Comment:
I introduce a small tricks here, so `query_word` will always input a larger limbs, and internally it will constrains new word expression is equals.
Any thought on this design :) ?

Reply:
Is this trick pushed in the PR branch?

Reply:
It looks a bit strange that `query_word` takes an argument.
Perhaps we could have these functions:
- query_word_unchecked
- query_word4_unchecked
- query_word16
- query_word32

And for gadgets that require word in lo/hi format with range checks we do:
```rust
let a = cb.query_word32();
Gadget::construct(a.to_word());
```
And for gadgets that don't require the check we do:
```rust
let a = cb.query_word_unchecked();
Gadget::construct(a.clone());
```

Another issue I find in the current form is that we allocate 2 extra cells that are not really needed, because the lo/hi can be expressions from the 16 lo limbs and 16 hi limbs.

Reply:
> Btw, where is `2 extra cells` :) ? I dont really get it

Ah that was before, when `qwery_word` took an argument.  It looked like this:
```
let word32 = cb.query_word32(); // Perform 32 queries
let word = cb.query_word(word32); // Perform 2 extra queries for lo/hi, and constraint them to be eq to the word32
```